cli/azd/internal/appdetect/docker.go

```diff
@@ -70,3 +82,50 @@ func parsePortsInLine(s string) ([]Port, error) {
 	}
 	return ports, nil
 }
+
+func addDefaultDockerfile(path string, hasParentPom bool) error {
+	if _, err := os.Stat(path); err != nil {
+		return fmt.Errorf("error accessing path %s: %w", path, err)
+	}
+
+	dockerfilePath := filepath.Join(path, "Dockerfile")
+	if _, err := os.Stat(dockerfilePath); err == nil {
+		fmt.Println("Dockerfile already exists, skipping creation.")
+		return nil
+	} else if !os.IsNotExist(err) {
+		return fmt.Errorf("error checking Dockerfile at path %s: %w", path, err)
+	}
+
+	file, err := os.Create(dockerfilePath)
+	if err != nil {
+		return fmt.Errorf("failed to create Dockerfile at %s: %w", dockerfilePath, err)
+	}
+	defer file.Close()
+
+	// for single-module project, we have to maven build first, then copy and run jar
+	// for multi-module project, just copy and run jar because of prepackage hook
+	var dockerfileContent string
+	if hasParentPom {
+		dockerfileContent = `FROM mcr.microsoft.com/openjdk/jdk:17-distroless
+COPY ./target/*.jar app.jar
+EXPOSE 8080
+ENTRYPOINT ["java", "-jar", "/app.jar"]`
```

Comment:
nit: Move this string out of func, and define a const, do like this will make code look better.

Reply:
Updated.